scrapi/migrations.py

```diff
@@ -17,57 +17,62 @@
-
-    if not kwargs.get('dry'):
-        tasks.process_raw(raw)
-        tasks.process_normalized(tasks.normalize(raw, raw['source']), raw)
-        logger.info('Processed document from {} with id {}'.format(doc.source, raw['docID']))
-
-        es.delete(index=settings.ELASTIC_INDEX, doc_type=doc.source, id=raw['docID'], ignore=[404])
-        es.delete(index='share_v1', doc_type=doc.source, id=raw['docID'], ignore=[404])
-
-    logger.info('Deleted document from {} with id {}'.format(doc.source, raw['docID']))
-
-
-@tasks.task_autoretry(default_retry_delay=1, max_retries=1)
-def renormalize(doc, **kwargs):
-    raw = RawDocument({
-        'doc': doc.doc,
-        'docID': doc.docID,
-        'source': doc.source,
-        'filetype': doc.filetype,
-        'timestamps': doc.timestamps,
-        'versions': doc.versions
-    })
-    if not kwargs.get('dry'):
-        tasks.process_normalized(tasks.normalize(raw, raw['source']), raw)
+    for doc in docs:
+        raw = RawDocument({
+            'doc': doc.doc,
+            'docID': doc.docID,
+            'source': target,
+            'filetype': doc.filetype,
+            'timestamps': doc.timestamps,
+            'versions': doc.versions
+        })
+
+        assert doc.source != target, "Can't rename {} to {}, names are the same.".format(doc.source, target)
+
+        if not kwargs.get('dry'):
+            tasks.process_raw(raw)
+            tasks.process_normalized(tasks.normalize(raw, raw['source']), raw)
+            logger.info('Processed document from {} with id {}'.format(doc.source, raw['docID']))
+
+            es.delete(index=settings.ELASTIC_INDEX, doc_type=doc.source, id=raw['docID'], ignore=[404])
+            es.delete(index='share_v1', doc_type=doc.source, id=raw['docID'], ignore=[404])
+
+        logger.info('Deleted document from {} with id {}'.format(doc.source, raw['docID']))
+
+
+@tasks.task_autoretry(default_retry_delay=1, max_retries=5)
+def renormalize(docs, *args, **kwargs):
+    for doc in docs:
+        raw = RawDocument({
+            'doc': doc.doc,
+            'docID': doc.docID,
+            'source': doc.source,
+            'filetype': doc.filetype,
+            'timestamps': doc.timestamps,
+            'versions': doc.versions
+        })
+        if not kwargs.get('dry'):
+            tasks.process_normalized(tasks.normalize(raw, raw['source']), raw)
 
 
 @tasks.task_autoretry(default_retry_delay=30, max_retries=5)
-def delete(doc, sources=None, **kwargs):
-    assert sources, "To run this migration you need a source."
-    doc.timeout(5).delete()
-    es.delete(index=settings.ELASTIC_INDEX, doc_type=sources, id=doc.docID, ignore=[404])
-    es.delete(index='share_v1', doc_type=sources, id=doc.docID, ignore=[404])
+def delete(docs, sources=None, **kwargs):
+    for doc in docs:
+        assert sources, "To run this migration you need a source."
+        doc.timeout(5).delete()
```

Comment:
If this task is retried, the first few docs will fail. If doc has already been deleted, we probably just want to skip it.